x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -615,6 +615,60 @@ private void clustersAndIndices(String command, String indexString1, String inde
         );
     }
 
+    public void testValidFromIndexPattern() {
+        var patterns = randomList(1, 5, () -> {
+            String pattern = randomIndexIdentifier();// index or alias
+            if (randomBoolean()) {// pattern
+                pattern += "*";
+            }
+            if (randomBoolean()) {// quoted
+                pattern = "\"" + pattern + "\"";
+            }
+            if (randomBoolean()) {// remote cluster
+                var cluster = randomIdentifier();
+                if (randomBoolean()) {// quoted
+                    cluster = "\"" + cluster + "\"";
+                }
+                pattern = cluster + ":" + pattern;
+            }
+            if (pattern.contains(":") && pattern.contains("\"") == false) {// quote entire "cluster:index"
+                pattern = "\"" + pattern + "\"";
+            }
+            return pattern;
+        });
+
+        var plan = statement("FROM " + String.join(",", patterns));
+        var expected = String.join(",", patterns).replace("\"", "");
+
+        assertThat(plan, instanceOf(UnresolvedRelation.class));
+        assertThat(((UnresolvedRelation) plan).table().index(), equalTo(expected));
+    }
+
+    private static String randomIndexIdentifier() {
+        // https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-path-params
+        var validFirstCharacters = "abcdefghijklmnopqrstuvwxyz0123456789!'$^&";
+        var validCharacters = validFirstCharacters + "+-_.";
+
+        var index = new StringBuilder();
+        if (randomInt(9) == 0) {// hidden index
+            index.append('.');
+        }
+        index.append(randomCharacterFrom(validFirstCharacters));
+        for (int i = 0; i < randomIntBetween(1, 100); i++) {
+            index.append(randomCharacterFrom(validCharacters));
+        }
+        return index.toString();
+    }
+
+    private static char randomCharacterFrom(String str) {
+        return str.charAt(randomInt(str.length() - 1));
+    }
+
+    public void testInvalidFromIndexPattern() {
+        expectError("FROM \"remote:\":index", "line 1:6: cluster string [remote:] must not contain ':'");
+        expectError("FROM \"remote:invalid\":index", "line 1:6: cluster string [remote:invalid] must not contain ':'");
+    }
+
     public void testInvalidQuotingAsFromIndexPattern() {
```

Comment:
Maybe we should also add tests for invalid quoting of the remote name itself for good measure.